runtime/memprof.c

```diff
@@ -183,7 +183,7 @@ static void rand_batch(void) {
   for(i = 0; i < RAND_BLOCK_SIZE; i++) {
     double f = B[i];
     CAMLassert (f >= 1);
-    if(f > Max_long) rand_geom_buff[i] = Max_long;
+    if(f > (double)Max_long) rand_geom_buff[i] = Max_long;
```

Comment:
Is this change intended?

Reply:
Yup, it fails on clang currently because of missing typing coercion and it seems better to properly do the type coercion instead of disabling the error.